### PR TITLE
fixes cli build on windows

### PR DESCRIPTION
**Pull request by fundies**
1) Deletes unused compiler artifacts designed for running pre and post build scripts.
2) Improves the CLI to detect the correct compileEGMf file on each operating system.
3) Updates the CLI to build for Microsoft Windows.

### CODE REVIEW

**Comment by RobertBColton** [comment]
@JoshDreamland can you also explain to me why 642ba867d5f3d170c857752818278525d65a14f8 forces the shell to be cmd and if we should perhaps undo that? Please review the discussion you had on the forum about this. As far as I understand, @fundies could not get the cross-compiler working in the CLI with it present and removing it seems to have no effect on my end.

File: CompilerSource/compiler/compile.cpp [hunk 1]
```diff
@@ -732,9 +732,9 @@ wto << "namespace enigma_user {\nstring shader_get_name(int i) {\n switch (i) {\
 
     // Restore the compilers original working directory.
     #if CURRENT_PLATFORM_ID == OS_WINDOWS
-    SetCurrentDirectory(prevdir);
+    SetCurrentDirectory(prevdir.data());
     #else
-    chdir(prevdir);
+    chdir(prevdir.data());
     #endif
```
Comment (RobertBColton) [inline CompilerSource/compiler/compile.cpp]: @JoshDreamland can you please explain to me, just for my own curiosity, what was wrong with this code? I tried digging to find why it was crashing but I don't see any. The only thing I can think of is that Travis CI deleted the current working directory when it was called or for some other reason it did not exist.

Comment (JoshDreamland) [reply CompilerSource/compiler/compile.cpp]: I am not sure that this fixed the issue, but the reason I told fundies to try it was that 4KiB is a lot of data to just throw on the stack. Typically the stack is about a megabyte in size, but I figured Travis might be running us with a much smaller version to conserve resources.

**Comment by JoshDreamland** [comment]
I'm not sure why they tried to force Make to use `cmd`; I think that was before we shipped MSys with ENIGMA. It seems they just tried to pass a variable called "SHELL" to Make, though, which will not actually do anything (GNU Make reads an *environment* variable called `SHELL` to determine which shell to use; setting a Make variable called "SHELL" will not change anything).

**Comment by RobertBColton** [comment]
Ok great, I agree with these changes then. I can delete that from Windows after I finish abstracting ``screen_redraw()`` then I can follow up on that by adding Appveyor to test Windows.

File: CommandLine/emake/EnigmaPlugin.cpp [hunk 2]
```diff
@@ -29,23 +29,28 @@ int EnigmaPlugin::Init()
...
 
-  _handle = dlopen(("./libcompileEGMf" + extension).c_str(), RTLD_LAZY);
+  std::string pluginName = "./" + prefix + "compileEGMf" + extension;
+  
+  _handle = dlopen(pluginName.c_str(), RTLD_LAZY);
 
   if (!_handle)
   {
-    std::cerr << "Error Loading Plugin" << std::endl;
+    std::cerr << "Error Loading Plugin " << pluginName << std::endl;
```
Comment (JoshDreamland) [inline CommandLine/emake/EnigmaPlugin.cpp]: Nit: Better to quote names, in case they're empty.

Comment (RobertBColton) [reply CommandLine/emake/EnigmaPlugin.cpp]: This is fine for now because he's only loading a single plugin, just remember to do it in the future.

**Pull request merged**